paramiko/proxy.py

```diff
@@ -105,7 +105,8 @@ def recv(self, size):
             raise ProxyCommandFailure(' '.join(self.cmd), e.strerror)
 
     def close(self):
-        os.kill(self.process.pid, signal.SIGTERM)
+        self.process.kill()
+        self.process.poll()
```

Comment:
I think `poll()` does clean up zombie processes, it just doesn't wait for the process if it hasn't exited yet. It's like libc `wait()` with `WNOHANG`. So in this case it was "too soon", it needed to check if the process did exit, and if not, try again.